rust/main/agents/relayer/src/msg/op_submitter.rs

```diff
@@ -1011,39 +1020,64 @@ async fn send_back_on_failed_submission(
 
 #[derive(Debug, Clone)]
 pub struct SerialSubmitterMetrics {
+    pub destination: String,
     pub(crate) submitter_queue_length: IntGaugeVec,
-    pub(crate) ops_prepared: IntCounter,
-    pub(crate) ops_submitted: IntCounter,
-    pub(crate) ops_confirmed: IntCounter,
-    pub(crate) ops_failed: IntCounter,
-    pub(crate) ops_dropped: IntCounter,
+    pub(crate) ops_prepared: IntCounterVec,
+    pub(crate) ops_submitted: IntCounterVec,
+    pub(crate) ops_confirmed: IntCounterVec,
+    pub(crate) ops_failed: IntCounterVec,
+    pub(crate) ops_dropped: IntCounterVec,
 }
 
 impl SerialSubmitterMetrics {
     pub fn new(metrics: impl AsRef<CoreMetrics>, destination: &HyperlaneDomain) -> Self {
         let destination = destination.name();
+
         Self {
+            destination: destination.to_string(),
             submitter_queue_length: metrics.as_ref().submitter_queue_length(),
-            ops_prepared: metrics
-                .as_ref()
-                .operations_processed_count()
-                .with_label_values(&["prepared", destination]),
-            ops_submitted: metrics
-                .as_ref()
-                .operations_processed_count()
-                .with_label_values(&["submitted", destination]),
-            ops_confirmed: metrics
-                .as_ref()
-                .operations_processed_count()
-                .with_label_values(&["confirmed", destination]),
-            ops_failed: metrics
-                .as_ref()
-                .operations_processed_count()
-                .with_label_values(&["failed", destination]),
-            ops_dropped: metrics
-                .as_ref()
-                .operations_processed_count()
-                .with_label_values(&["dropped", destination]),
+            ops_prepared: metrics.as_ref().operations_processed_count(),
+            ops_submitted: metrics.as_ref().operations_processed_count(),
+            ops_confirmed: metrics.as_ref().operations_processed_count(),
+            ops_failed: metrics.as_ref().operations_processed_count(),
+            ops_dropped: metrics.as_ref().operations_processed_count(),
+        }
+    }
+
+    pub fn inc_prepared_with_app_context(&self, app_context: Option<String>) {
```

Comment:
nit: I don't feel strong but kinda feels like the `_with_app_context` is a little redundant / stylistically noisy, wdyt of just changing all these to `inc_prepared`, `inc_submitted`, etc? we already get the app_context info with the param name

Reply:
yeah good idea